go-selinux/selinux_linux.go

```diff
@@ -1095,3 +1104,115 @@ func dupSecOpt(src string) ([]string, error) {
 func disableSecOpt() []string {
 	return []string{"disable"}
 }
+
+func getDefaultContextFromReaders(c defaultSECtx) (string, error) {
+	if c.verifier == nil {
+		return "", fmt.Errorf("verifier function must not be nil")
+	}
+
+	context, err := newContext(c.scon)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to create label for %s", c.scon)
+	}
+
+	findUserInContext := func(r io.Reader) string {
+		fromRole := context["role"]
+		fromType := context["type"]
+		bufin := bufio.NewReader(r)
+
+		for done := false; !done; {
```

Comment:
I think it will be easier/cleaner to use bufio scanner here:
```go
s := bufio.NewScanner(r)
for scanner.Scan() {
        line = strings.TrimSpace(s.Text())
        ...
}
if err := scanner.Err(); err != nil {
        // FIXME log/return err?
        return ""
}

Reply:
BTW feel free to rewrite `readConfig` (from which you apparently copy-pasted this code) to use bufio.Scanner, too (in a separate commit).